localops/src/process.rs

```diff
@@ -0,0 +1,66 @@
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+
+use anyhow::{anyhow, Result};
+#[cfg(any(target_os = "macos", target_os = "linux"))]
+use nix::sys::signal::{kill, Signal};
+#[cfg(any(target_os = "macos", target_os = "linux"))]
+use nix::unistd::Pid;
+
+pub fn daemonize(bin_path: PathBuf) -> Result<()> {
+    daemonize_process(bin_path)
+}
+
+#[cfg(not(any(target_os = "macos", target_os = "linux", target_os = "windows")))]
+fn daemonize_process(bin_path: PathBuf) -> Result<()> {
+    Err(anyhow!(
+        "deamonizing processes is not supported on your operating system"
+    ))
+}
+
+#[cfg(any(target_os = "macos", target_os = "linux"))]
+fn daemonize_process(bin_path: PathBuf) -> Result<()> {
+    // ¯\_(ツ)_/¯
+    let status = Command::new("bash")
+        .arg("-c")
+        // TODO(b5): might be nice to capture output in a log file at some point?
+        .arg(format!(
+            "nohup {} > /dev/null 2>&1 &",
+            bin_path.to_str().unwrap(),
+        ))
+        .stderr(Stdio::null())
+        .stdout(Stdio::null())
+        .status()?;
+
+    if !status.success() {
+        Err(anyhow::anyhow!("couldn't daemonize binary"))?;
+    }
+    Ok(())
+}
+
+#[cfg(target_os = "windows")]
+fn daemonize_process(bin_path: PathBuf) -> Result<()> {
+    Err(anyhow!("deamonizing processes on windows is not supported"))
+}
+
+pub fn stop(pid: u32) -> Result<()> {
+    stop_process(pid)
+}
+
+#[cfg(not(any(target_os = "macos", target_os = "linux", target_os = "windows")))]
+fn stop_process(pid: u32) -> Result<()> {
+    Err(anyhow!(
+        "stopping processes is not supported on your operating system"
+    ))
+}
+
+#[cfg(any(target_os = "macos", target_os = "linux"))]
+fn stop_process(pid: u32) -> Result<()> {
```

Comment:
please make all apis that use `u32`s currently, use a new type `Pid`, that looks sth like this

```rust
#[cfg(unix)]
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct Pid(nix::Pid);
#[cfg(not(unix))]
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct Pid; // TODO: fill in for each plattform when supported

#[cfg(unix)]
impl From nix::Pid for Pid {
 // .. 
}

impl std::fmt::Display for Pid {
 // ..
}
```

Reply:
I've marked this as a TODO for later